HOWTO-make-a-release.md

```diff
@@ -0,0 +1,378 @@
+including the "Next Steps" at the end of that page.
+
+[creating a security advisory]:
+<https://docs.github.com/en/free-pro-team@latest/github/managing-security-vulnerabilities/creating-a-security-advisory>
+
+The release person and the reviewer are allowed to use other means to
+share the commits to be reviewed if they desire.
+
+The release person and the reviewer must have a conversation to confirm
+or figure out how review shall be done.
+
+# Pre-publishing tasks
+
+Some of the actions in this section need to be repeated for each OpenSSL
+version released.
+
+## Freeze the source repository
+
+The day before the release, freeze the main repository.  This locks
+out everyone but the named user, who is doing the release, from doing
+any pushes.  Someone other than the person doing the release should
+run the command.  For example:
+
+    ssh openssl-git@git.openssl.org freeze openssl NAME
+
+## Prepare your repository checkouts
+
+You will need to checkout at least three working trees:
+
+-   one for the website
+
+        git clone openssl-git@git.openssl.org:openssl-web.git website
+
+-   one for extra tools
+
+        git clone openssl-git@git.openssl.org:openssl-tools.git tools
+
+    The resulting directory will be referred to as `$TOOLS`
+
+-   At least one for openssl source
+
+        git clone openssl-git@git.openssl.org:openssl.git
+
+    If you're doing multiple releases in one go, there are many ways to deal
+    with it.  One possibility, available since git 2.5, is to use `git
+    worktree`:
+
+        (cd openssl;
+         git worktree add ../openssl-1.1.1 OpenSSL_1_1_1-stable)
+
+## Make sure that the openssl source is up to date
+
+The person doing the release and the reviewer should both sanity-check
+the source to be released at this point.  Checks to consider include
+building and verify that make test passes on multiple plaforms - Linux,
+Windows, etc.
+
+For each source checkout, make sure that the CHANGES.md / CHANGES and
+NEWS.md / NEWS files have been updated and reviewed.
+
+NEWS.md / NEWS should contain a summary of any changes for the release,
+and for a security release is (often just a list of the CVEs addressed).
+You should also update NEWS in the master branch to include details of
+all releases.  Just update the NEWS bullet points - do not change the
+release date, keep it as **under development**.
+
+Add any security fixes to the tree and commit them.
+
+Make sure that the copyrights are updated.  This script will update
+the copyright markers and commit the changes (where $TOOLS stands for
+the openssl-tools.git checkout directory):
+
+    $TOOLS/release-tools/do-copyright-year
+
+Obtain approval for these commits from the reviewer and add the
+reviewed-by headers as required.
+
+*Do not push* changes to the public repo at this stage.
+(the public repo being `openssl-git@git.openssl.org:openssl.git`)
+
+*Do* send the auto-generated commits to the reviewer and await their
+approval.
```

Comment:
If there are no security fixes then there should be no problem with pushing CHANGES/NEWS updates and copyright year updates - and in fact I usually do. Obviously if there are security fixes then that changes things.

Reply:
Yes?

Reply:
Probably I made this comment against the wrong line. I meant to point out that the "Do not push changes to the  public repo" line is not strictly accurate.

Reply:
If you read the parenthesis as well, it *is* accurate.

Reply:
I can see how the nomenclature is confusing, though.

I've used `final` as a remote in my repos, to have a uniform way for a command line that does the final merge, perhaps that would be a fitting nomenclature?  Or perhaps just being more specific, saying "public openssl repo" rather than the mere "public repo"?

Reply:
> If you read the parenthesis as well, it is accurate.

Err no, its not. If we're not doing a security release then you *can* push CHANGES/NEWS and do-copyright year updates (which is the opposite of what the instructions state), and I usually do. If security fixes are involved then you can't

Reply:
Ah, you mean like that...  yeahok.  I usually wait with that final push 'til "Publish the release" further down...  But OK, I'll rephrase that a bit

Reply:
Fixed